pycsw/ogc/api/records.py

```diff
@@ -1215,6 +1217,13 @@ def record2json(record, url, collection, mode='ogcapi-records'):
                     })
                 except Exception as err:
                     LOGGER.exception(f"failed to parse contact of {record.identifier}: {err}")
+            for r2 in "creator,publisher,contributor".split(","): # match role-fields with contacts
```

Comment:
I think `record` is an object here (not a dict)?  Perhaps the below can work?

```python
            for r2 in ['creator', 'publisher', 'contributor']:  # match role-fields with contacts
                if r2 not in roles and getattr(record, r2, None) not in [None, '']:
                    rcnt.append({
                        'organization': getattr(record, r2),
                        'roles': [r2]
                    })
```

Reply:
thanx tom, you're right, now fixed and tested